packages/graphback-runtime/src/service/DefaultCRUDService.ts

```diff
@@ -1,124 +1,130 @@
+            const topic = subscriptionTopicMapping(GraphbackOperationType.UPDATE, name);
+            const payload = this.buildEventPayload('updated', name, result);
             await this.pubSub.publish(topic, payload);
         }
 
         return result;
     }
 
     // tslint:disable-next-line: no-reserved-keywords
-    public async delete(inputType: InputModelTypeContext, id: string, context?: GraphbackContext): Promise<string> {
-        this.logger.log(`deleting object ${inputType.name}`)
+    public async delete(name: string, id: string, data?: T, context?: GraphbackContext): Promise<string> {
+        this.logger.log(`deleting object ${name}`)
 
-        const result = await this.db.delete(inputType, id, context);
-        if (this.pubSub && inputType.config.subDelete) {
-            const topic = subscriptionTopicMapping(GraphbackOperationType.DELETE, inputType.name);
-            const payload = this.buildEventPayload('deleted', inputType, result);
+        const result = await this.db.delete(name, id, data, context);
+        if (this.pubSub && context && context.publish) {
+            const topic = subscriptionTopicMapping(GraphbackOperationType.DELETE, name);
+            const payload = this.buildEventPayload('deleted', name, result);
             await this.pubSub.publish(topic, payload);
         }
 
         return result;
     }
 
-    public read(inputType: InputModelTypeContext, id: string, context?: GraphbackContext): Promise<T> {
-        this.logger.log(`reading object ${inputType.name}`)
+    public read(name: string, id: string, context?: GraphbackContext): Promise<T> {
+        this.logger.log(`reading object ${name}`)
 
-        return this.db.read(inputType, id, context);
+        return this.db.read(name, id, context);
     }
 
-    public findAll(inputType: InputModelTypeContext, context?: GraphbackContext): Promise<T[]> {
-        this.logger.log(`querying object ${inputType.name}`)
+    public findAll(name: string, context?: GraphbackContext): Promise<T[]> {
+        this.logger.log(`querying object ${name}`)
 
-        return this.db.findAll(inputType, context);
+        return this.db.findAll(name, context);
     }
 
     // tslint:disable-next-line: no-any
-    public findBy(inputType: InputModelTypeContext, filter: any, context?: GraphbackContext): Promise<T[]> {
-        this.logger.log(`querying object ${inputType.name} with filter ${JSON.stringify(filter)}`)
+    public findBy(name: string, filter: any, context?: GraphbackContext): Promise<T[]> {
+        this.logger.log(`querying object ${name} with filter ${JSON.stringify(filter)}`)
 
-        return this.db.findBy(inputType, filter, context);
+        return this.db.findBy(name, filter, context);
     }
 
-    public subscribeToCreate(inputType: InputModelTypeContext, context?: GraphbackContext): AsyncIterator<T> | undefined {
+    public subscribeToCreate(name: string, context?: GraphbackContext): AsyncIterator<T> | undefined {
```

Comment:
`context` is not used here. Will it be?

Reply:
In generic scenario layers can support filtering and other capabilities. So I think that it should be there despite this particular implementation not benefiting from it.